core/Pos/Binary/Core/Script.hs

```diff
@@ -1,37 +1,104 @@
+{-# LANGUAGE TemplateHaskell #-}
 module Pos.Binary.Core.Script () where
 
 import           Universum
 
-import qualified Data.Store         as Store
 import qualified PlutusCore.Program as PLCore
 import qualified PlutusCore.Term    as PLCore
+import qualified PlutusTypes.ConSig as PLTypes
+import qualified PlutusTypes.Type   as PLTypes
+import qualified Utils.ABT          as ABT
+import qualified Utils.Names        as Names
+import qualified Utils.Vars         as Vars
 
-import           Pos.Binary.Class   (Bi (..), UnsignedVarInt (..), label, labelP, labelS,
-                                     putField)
+import           Data.Hashable      (Hashable, hashWithSalt)
+import           Data.SafeCopy      (SafeCopy(..))
+import           Pos.Binary.Class   (Bi (..), deriveSimpleBi, Cons(..), Field(..),
+                                     serialize', getCopyBi, putCopyBi, genericEncode, genericDecode)
 import           Pos.Core.Script    ()
-import           Pos.Core.Types     (Script (..))
+import           Pos.Core.Types     (Script (..), ScriptVersion)
 
-instance Bi PLCore.Term where
-    size = Store.size
-    {-# INLINE size #-}
-    get = label "PLCore.Term" $ Store.peek
-    {-# INLINE get #-}
-    put = labelP "PLCore.Term" . Store.poke
-    {-# INLINE put #-}
+
+instance Bi PLCore.Term => SafeCopy PLCore.Term where
```

Comment:
Here I had to write almost all generically-derived instances as it was the only sensible thing to do, and it's no worse that the old `store` implementation (which AFAIK it uses `Generic` as well for the default case).